test/integration/ggrc/services/test_query/test_basic.py

```diff
@@ -1311,46 +1494,144 @@ def test_multiple_ca_sorting(self):
 
   def test_ca_query_eq(self):
     """Test CA date fields filtering by = operator."""
-    date = datetime(2015, 5, 18)
+    ca_values = [date(2015, 5, 16),
+                 date(2015, 5, 17),
+                 date(2015, 5, 18),
+                 date(2015, 5, 19),
+                 date(2015, 5, 20),
+                 ]
+    program_slugs = []
+    expected_date = datetime(2015, 5, 18)
+
+    with factories.single_commit():
+      for i in range(5):
+        program = factories.ProgramFactory(title="program {}".format(i))
+        program_slugs.append(program.slug)
+
+    ca_date = db.session.query(CAD).filter_by(title="CA date").first()
+    for program_slug, ca_value in zip(program_slugs, ca_values):
+      pgm = db.session.query(all_models.Program).filter_by(
+          slug=program_slug).first()
+      factories.CustomAttributeValueFactory(
+          custom_attribute=ca_date,
+          attributable=pgm,
+          attribute_value=ca_value,
+      )
```

Comment:
Please consider rewriting this piece of code in a more clear manner. I think something like the following will do. Besides it allows to avoid unnecessary DB query for program which is currently being performed in the second for loop.
```suggestion
    expected_date = datetime(2015, 5, 18)

    ca_date = db.session.query(CAD).filter_by(title="CA date").one()
    with factories.single_commit():
      for i, ca_value in enumerate(ca_values):
        program = factories.ProgramFactory(title="program {}".format(i))
        factories.CustomAttributeValueFactory(
            custom_attribute=ca_date,
            attributable=program,
            attribute_value=ca_value,
        )
```